lib/rubocop/cop/layout/empty_lines_around_arguments.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Layout
+      # This cops checks if empty lines exist around the arguments
+      # of a method invocation.
+      #
+      # @example
+      #   # bad
+      #   do_something(
+      #     foo
+      #
+      #   )
+      #
+      #   # good
+      #   do_something(
+      #     foo
+      #   )
+      #
+      class EmptyLinesAroundArguments < Cop
+        MSG = 'Empty line detected around arguments.'.freeze
+
+        def on_send(node)
+          return if node.single_line?
+          return if empty_lines(node).empty?
+          extra_lines(node) do |range|
+            add_offense(node, location: range)
+          end
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            extra_lines(node) do |range|
+              corrector.remove(range)
+            end
+          end
+        end
+
+        def empty_lines(node)
+          @empty_lines ||= begin
```

Comment:
I'm not sure, but I think memoization might mess with the correction loop in some cases.

Reply:
I just tried removing it. And for some reason `rake interal_investigation` then returns offenses:

```
1058 files inspected, 25 offenses detected
RuboCop failed!
```

I got the above results consistently 3 times in a row, only on 3 files - `spec/rubocop/cop/layout/extra_spacing_spec.rb`, `spec/rubocop/cop/layout/indent_array_spec.rb`, & `spec/rubocop/cop/offense_spec.rb`.

Adding the memoization back lets the investigation consistently pass.

Any ideas? I'll see if I can figure it out, but it doesn't make much sense at first inspection.

Reply:
Possibly my cache is corrupt somehow?